### DISCUSSION TITLE
How to choose from images list for <img> edition?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#902] [Q&A] [answered]
Hi

Is there a way that `<img>` edition links to a list of url/images (instead of Mavo popup)?

I would like to be able to choose from a list of a preloaded images (referenced inside a json file for example), and allow user to choose from them, instead of providing a url or uploading it each time

As described on documentation [here](https://mavo.io/docs/properties#mv-editor), 
I tried to apply it to `<img>` element: see my `[codepen](https://codepen.io/galinhalx/pen/KKoxWeX)`

but <img> editor always popup mavo's window to uplad from file

Does someone experienced that, or find out a solution to this usecase?

**Comment by DmitrySharabin** [comment, marked as answer]
Hey there, @GalinhaLX! 👋🏻

Do you mean something like that:

```html
<img property="myImage" mv-editor="#imageList" mv-default="https://mavo.io/img/logo-mit.svg" />

<select id="imageList" hidden>
	<option>https://i.imgur.com/LPeWcag.png</option>
	<option>https://mavo.io/logo.svg</option>
	<option>https://mavo.io/img/logo-mit.svg</option>
</select>
```

Please note that to replace the default editor, we should use the `mv-editor` attribute. And the value of this attribute is a valid selector (like in CSS) that allows Mavo to find the corresponding element on the page. That's why we use `#imageList` in the code above (since we ask Mavo to find the `<select>` element by its `id` attribute).

Hope it helps!

**Reply by GalinhaLX** [reply]
Thanks @DmitrySharabin 

Seems I missunderstood mv-editor-* (still not sure to understand that one by the way)

**Reply by DmitrySharabin** [reply]
Consider the following code: `<span property="email">jhon@doe.com</span>`. By default, Mavo will use `<input type="text">` to edit this property. But what we relly need is to use `<input type="email">` as the editor (for built-in email validation and better UX).

But how can we change the value of the `type` attribute of the property editor? The `mv-editor-*` family of attributes allow us to pass attributes and/or their values to the property editor. All we need to do is just replace `*` with the name of the needed attribute. And give this attribute the needed value. So, in our case, we would have: `mv-editor-type="email"`. The full code snippet would look like that: `<span property="email" mv-editor-type="email">jhon@doe.com</span>`.

We can customize the property editor even further. Suppose we want to limit allowed emails to the `globex.com` domain. The `pattern` attribute is for the resque. And once again, we can pass this attribute and its value to the property editor with the `mv-editor-*` attribute: `<span property="email" mv-editor-type="email" mv-editor-pattern=".+@globex\.com">jhon@doe.com</span>`.

**Comment by karger** [comment]
While I may not be interpreting your question correctly, the following 
two things should work:

1) define a chosenImage property, place <img src="chosenImage"> in your 
app, and use a select menu to let the user select a value for chosenImage

2) if you want them to select an image by clicking on a thumbnail, you 
can put an mv-action on each thumbnail that uses set() to change the 
value of chosenImage based on which thumbnail they click.

**Reply by GalinhaLX** [reply]
Hi @karger 

Thanks for both anwers. I finally used 2nd option
